model.frg

```diff
@@ -119,15 +119,25 @@ pred safeEndPoint[s : State] {
 -----3 levels of passwords------
 pred unsafePassword[s : State] {
     s.user.password.length < 3 or
-
+    (s.user.password.hasSpecChars = False and s.user.password.hasNumbers = False and s.user.password.hasUpperCase = False) or
+    s.user.password.hasPattern = True
 }
 
 pred semisafePassword[s : State] {
-    s.user.password 
+    s.user.password.length = 3 
```

Comment:
Is there a reason we're choosing 3 here, in particular?

Reply:
Nothing special, just the way I chose to normalize it. So it needs 4 to be considered secure
--multiples of 5, 1 = (1-5) chars, 2 = (5-10) chars, 3 = (10-15), 4 = (15-MAX)